pipelines/pepatac.py

```diff
@@ -759,7 +774,7 @@ def main():
                 unmap_fq1, unmap_fq2 = _align_with_bt2(
                     args, tools, args.paired_end, True,
                     unmap_fq1, unmap_fq2, reference,
-                    assembly_bt2=_get_bowtie2_index(res.genomes, reference),
+                    assembly_bt2=_get_bowtie2_index(res.rgc, reference),
```

Comment:
these should be using the `res` variable instead of re-deriving the index

Reply:
these calls appear to be for prealignment assemblies, not the main `args.genome_assembly` from which the relevant bowtie value on the `res` object is derived.

Reply:
ah, got it. Can these just use `rgc.get_asset(reference, "bowtie2_index")` ?

Reply:
Sure, it can; why’s that better? It seems both worse in three ways—more extensive use of a constant, slightly less readable (debatably, admittedly), and fixates on an implementation in more places rather than routing through one (the fxn implementation)

Reply:
It's better because it will be the same use mode in every pipeline or script that uses refgenconf. Sure, we could define local functions in each script to do every task that we use an external tool for...but that in some ways ruins the point of the shared way of doing something. we also lose the parallelism even *within this pipeline* with all other refgenie assets this way.

I also think putting the "constant" into a function name is not as good because to me at least that's not *really* a constant... it's more a variable representing an asset. we query lots of assets this way.  we could define "get_bowtie2_index" on the rgc object but then we have to do that for every asset....

`get_bowtie2_index` also adds no functionality outside of `get_asset`. it has effectively been abstracted away and is no longer necessary.  both are just function calls... one is a function call that wraps no additional functionality over another function call. the complexity layer adds nothing. we effectively replaced the local function with a more universal function call and we should use that.